compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -12025,4 +12043,338 @@ private STNode parseByteArrayContent(SyntaxKind kind) {
+     * @return Parsed node
+     */
+    private STNode parseXMLStepExtendList() {
+        List<STNode> xmlStepExtendList = new ArrayList<>();
+        STToken nextToken = peek();
+
+        // Return an empty list
+        if (isEndOfXMLStepExtend(nextToken.kind)) {
+            return STNodeFactory.createNodeList(xmlStepExtendList);
+        }
+
+        nextToken = peek();
+        STNode xmlStepExtend;
+        while (!isEndOfXMLNamePattern(nextToken.kind)) {
+            xmlStepExtend = parseXMLStepExtend();
+            xmlStepExtendList.add(xmlStepExtend);
+            nextToken = peek();
+        }
+
+        return STNodeFactory.createNodeList(xmlStepExtendList);
+    }
+
+    private boolean isEndOfXMLStepExtend(SyntaxKind tokenKind) {
+        switch (tokenKind) {
+            case DOT_LT_TOKEN:
+            case OPEN_BRACKET_TOKEN:
+            case DOT_TOKEN:
+                return false;
+            default:
+                return true;
+        }
+    }
+
+    /**
+     * Parse xml step extend list.
+     * <p>
+     * <code>
+     * xml-step-extend :=
+     *    .< xml-name-pattern >
+     *    | [ expression ]
+     *    | . method-name ( arg-list )
+     * </code>
+     *
+     * @return Parsed node
+     */
+    private STNode parseXMLStepExtend() {
+        STToken token = peek();
+        switch (token.kind) {
+            case DOT_LT_TOKEN:
+                return parseXMLFilterExpressionRhs();
+            case DOT_TOKEN:
+                return parseMethodCallChain();
+            // token kind is already validated in isEndOfXMLStepExtend and reach here.
+            // therefore, a recover solution is not required for the default case.
+            case OPEN_BRACKET_TOKEN:
+            default:
+                return parseOpenBracketExpressionChain();
+
+        }
+    }
+
+    /**
+     * Parse open bracket expression chain.
+     * <p>
+     * <code>open-bracket-expression-chain := `[` expression `]`</code>
+     *
+     * @return Parsed node
+     */
+    private STNode parseOpenBracketExpressionChain() {
+        STNode openBracketToken = parseOpenBracket();
+        STNode expression = parseExpression();
+        STNode closeBracketToken = parseCloseBracket();
+        return STNodeFactory.createOpenBracketExpressionChainingNode(openBracketToken, expression, closeBracketToken);
+    }
+
+    /**
+     * Parse method call chain.
+     * <p>
+     * <code>method-call-chain := `.` method-name `(` arg-list `)`</code>
+     *
+     * @return Parsed node
+     */
+    private STNode parseMethodCallChain() {
+        STNode dotToken = parseDotToken();
+        STNode methodName = parseIdentifier(ParserRuleContext.FIELD_OR_FUNC_NAME);
+        STNode openParen = parseOpenParenthesis(ParserRuleContext.ARG_LIST_START);
+        STNode args = parseArgsList();
+        STNode closeParen = parseCloseParenthesis();
+        return STNodeFactory.createMethodCallChainingNode(dotToken, methodName, openParen, args, closeParen);
```

Comment:
Probably be able to simply let the `parseExpressionRhs` handles it, similar to above.